### PR TITLE
Adding logs to surface AZ-CLI and powershell errors

### PR DESCRIPTION
**Pull request by BALAGA-GAYATRI**
Screenshot 2021-11-11 at 5 50 02 PM

### CODE REVIEW

File: lib/main.js [hunk 1]
```diff
@@ -180,12 +208,11 @@ function main() {
         }
         catch (error) {
             if (!isAzCLISuccess) {
-                core.error("Az CLI Login failed. Please check the credentials. For more information refer https://aka.ms/create-secrets-for-GitHub-workflows");
+                core.setFailed("Az CLI Login failed. Please check the credentials. For more information refer https://aka.ms/create-secrets-for-GitHub-workflows");
```
Comment (kaverma) [inline lib/main.js:211]: what's the difference between .error and .setFailed

File: src/main.ts [hunk 2]
```diff
@@ -10,6 +11,23 @@ var azPSHostEnv = !!process.env.AZUREPS_HOST_ENVIRONMENT ? `${process.env.AZUREP
...
     try {
+        //Options for error handling
+        let commandStdErr = false;
+        const loginOptions: ExecOptions = {
+            silent: true,
+            ignoreReturnCode: true,
+            failOnStdErr: true,
+            listeners: {
+                stderr: (data: Buffer) => {
+                    let error = data.toString().replace('ERROR','');
```
Comment (kaverma) [inline src/main.ts:22]: we want to replace only the starting ERROR substring and not all ERROR mentions right?

File: src/main.ts [hunk 3]
```diff
@@ -10,6 +11,23 @@ var azPSHostEnv = !!process.env.AZUREPS_HOST_ENVIRONMENT ? `${process.env.AZUREP
...
+                    if(error && error.trim().length !== 0)
```
Comment (kaverma) [inline src/main.ts:23]: nit: formatting - space after if

**Review by kaverma — APPROVED** [review]

File: lib/main.js (hunk 1 above)
Comment (BALAGA-GAYATRI) [reply lib/main.js:211]: execution stops here if `setFailed` otherwise it continues.

File: src/main.ts (hunk 3 above)
Comment (BALAGA-GAYATRI) [reply src/main.ts]: done

File: src/main.ts (hunk 2 above)
Comment (BALAGA-GAYATRI) [reply src/main.ts]: Made small change. It replaces only first occurrence in the beginning of the message  now.

**Pull request merged**